mds/models.py

```diff
@@ -170,12 +173,10 @@ def gps_point_as_geojson(self):
 
 class EventRecord(models.Model):
     timestamp = models.DateTimeField(db_index=True)
-    point = gis_models.PointField(null=True)
+    point = gis_models.PointField(blank=True, null=True)
     saved_at = models.DateTimeField(default=utils.timezone.now)
-    source = models.CharField(
-        choices=enums.choices(enums.EVENT_SOURCE),
-        default=enums.EVENT_SOURCE.push.name,
-        max_length=16,
+    source = UnboundedCharField(
```

Comment:
Note you're swapping a `character varying(16)` for a `text` on 1M+ lines. No idea how long it will take but it will lock the whole table, doesn't it?

Reply:
I'm unsure about this : PostgreSQL docs hints that this is the same underlying field, so this should (in theory) just amount to a constraint drop (hence no lock).

Reply:
I'll test on our biggest production dump.